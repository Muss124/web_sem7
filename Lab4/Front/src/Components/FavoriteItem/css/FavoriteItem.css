.FavoriteItem .WeatherTop {
    padding        : 15px;
    display        : flex;
    justify-content: space-between;
}

.FavoriteItem {
    width     : 45%;
    text-align: justify;
}