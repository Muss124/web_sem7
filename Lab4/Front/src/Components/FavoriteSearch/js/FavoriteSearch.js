import React from "react"
import store from "../../../store"
import '../css/FavoriteSearch.css'
import FetchCityData from '../../../Actions/FetchCityData'
import RefreshCity from "../../../Actions/RefreshCity"

export class FavoriteSearch extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        this.addGeo(this.state.value);
        event.preventDefault();
    }

    addGeo(value) {
        console.log("form submited");
        store.dispatch({ type: "FAVORITE_ADD", payload: { CityName : value } });
        store.dispatch(FetchCityData(value));
    }
    render() {
        return (
            <div className="searchform">
                <div>
                    <h1>Favorite</h1>
                </div>
                <div className="searchformInside">
                    <form onSubmit={this.handleSubmit}>
                        <input type="search" value={this.state.value} onChange={this.handleChange}></input>
                        <button type="submit" className="btn btn-secondary btn-circle" >Add</button>
                    </form>
                    <button className="btn btn-secondary" onClick={() => {
                        console.log("REFRESH");
                        var refData = store.getState();
                        refData.forEach(el => {
                            console.log(el);
                            store.dispatch(RefreshCity(el.CityName));
                        });
                    }} >&#8634;</button>
                </div>
            </div>
        );
    }
}