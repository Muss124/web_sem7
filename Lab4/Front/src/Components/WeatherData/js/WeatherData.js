import React from "react"
import store from "../../../store"
import "../css/WeatherData.css"
import ServerDeleteCity from '../../../Actions/ServerDeleteCity'

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		//console.log("here");
		//console.log(WeatherData);
		//console.log(this.props.isMain);
		if (this.props.isMain) {
			return (
				<div className="WeatherData">
					<div className="WeatherTop">
						<h2>{WeatherData.CityName}</h2>
						<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
						<h1>{WeatherData.Temperature} &#8451;</h1>
					</div>
					<div className="WeatherMain">
						<div className="WeatherMainEl">
							<div className="WeatherParam">Weather</div>
							<div className="WeatherValue">{WeatherData.Weather}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Humidity (%)</div>
							<div className="WeatherValue">{WeatherData.Humidity}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Pressure (hPa)</div>
							<div className="WeatherValue">{WeatherData.Pressure}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Wind speed (meter/sec)</div>
							<div className="WeatherValue">{WeatherData.Wind}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Longitude </div>
							<div className="WeatherValue">{WeatherData.Longitude}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Latitude</div>
							<div className="WeatherValue">{WeatherData.Latitude}</div>
						</div>
					</div>
				</div>
			);
		}
		else {
			return (
				<div className="WeatherData">
					<div className="WeatherTop">
						<h2>{WeatherData.CityName}</h2>
						<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
						<h1>{WeatherData.Temperature} &#8451;</h1>
						<button type="submit" className="btn btn-secondary" onClick={() => {
							console.log("REMOVE");
							//store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData.CityName });
							store.dispatch(ServerDeleteCity(WeatherData.CityName));
						}} >&#10006;</button>
					</div>
					<div className="WeatherMain">
						<div className="WeatherMainEl">
							<div className="WeatherParam">Weather</div>
							<div className="WeatherValue">{WeatherData.Weather}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Humidity (%)</div>
							<div className="WeatherValue">{WeatherData.Humidity}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Pressure (hPa)</div>
							<div className="WeatherValue">{WeatherData.Pressure}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Wind speed (meter/sec)</div>
							<div className="WeatherValue">{WeatherData.Wind}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Longitude </div>
							<div className="WeatherValue">{WeatherData.Longitude}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Latitude</div>
							<div className="WeatherValue">{WeatherData.Latitude}</div>
						</div>
					</div>
				</div>
			);
		}

	}
}