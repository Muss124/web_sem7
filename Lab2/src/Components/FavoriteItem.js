import React from 'react';
import {WeatherData} from "./WeatherData.js"

export class FavoriteItem extends React.Component {
	render() {
		
		if (this.props.value.loading) {
			return (
				<div>
					{this.props.value.city}
					<div class="spinner-border text-danger" role="status">
						<span class="sr-only">Loading...</span>
					</div>
				</div>
			);
		}
		else {
			return (
				<div class="FavoriteItem">
					<WeatherData value={this.props.value.data} />
				</div>
			);
		}
	}
}