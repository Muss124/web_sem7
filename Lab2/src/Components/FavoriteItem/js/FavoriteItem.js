import React from 'react';
import {WeatherData} from "../../WeatherData/js/WeatherData"
import '../css/FavoriteItem.css'


export class FavoriteItem extends React.Component {
	render() {
		console.log(this.props.value.data);
		console.log(this.props.value.data.length === 0);
		if (this.props.value.data.length === 0) {
			return (
				<div>
					{this.props.value.city}
					<div className="spinner-border text-danger" role="status">
						<span className="sr-only">Loading...</span>
					</div>
				</div>
			);
		}
		else {
			return (
				<div className="FavoriteItem">
					<WeatherData value={this.props.value.data} isMain={false}/>
				</div>
			);
		}
	}
}