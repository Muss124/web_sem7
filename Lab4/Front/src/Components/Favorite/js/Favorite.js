import React from 'react';
import { FavoriteSearch } from "../../FavoriteSearch/js/FavoriteSearch"
import { connect } from 'react-redux'
import { FavoriteItem } from "../../FavoriteItem/js/FavoriteItem"
import '../css/Favorite.css'
import FetchFavorite from "../../../Actions/FetchFavorite"
import store from "../../../store"

class Favorite extends React.Component {
    constructor(props) {
        super(props);
        store.dispatch(FetchFavorite());
    }
    render() {
        return (
            <div>
                <FavoriteSearch />
                <div className="FavoriteContainer" >
                    {
                        this.props.data.map((favCity, i = 0) => {
                            i++;
                            return ([

                                <FavoriteItem value={favCity} key={i} />

                            ])
                        })
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);