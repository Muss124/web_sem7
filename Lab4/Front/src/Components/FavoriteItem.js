import React from 'react';
import {WeatherData} from "./WeatherData.js"

export class FavoriteItem extends React.Component {
	render() {
		if (typeof this.props.value.loading != "boolean") {
			this.props.value = {
				loading : false, 
				city: this.props.value.CityName,
				data: this.props.value
			}
		}
		if (this.props.value.loading) {
			return (
				<div>
					{this.props.value.city}
					<div class="spinner-border text-danger" role="status">
						<span class="sr-only">Loading...</span>
					</div>
				</div>
			);
		}
		else {
			return (
				<div class="FavoriteItem">
					<WeatherData value={this.props.value.data} />
				</div>
			);
		}
	}
}