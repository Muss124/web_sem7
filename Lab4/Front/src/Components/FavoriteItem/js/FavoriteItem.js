import React from 'react';
import {WeatherData} from "../../WeatherData/js/WeatherData"
import '../css/FavoriteItem.css'


export class FavoriteItem extends React.Component {
	render() {
		if (Object.keys(this.props.value).length === 1) {
			return (
				<div>
					{this.props.value.CityName}
					<div className="spinner-border text-danger" role="status">
						<span className="sr-only">Loading...</span>
					</div>
				</div>
			);
		}
		else {
			return (
				<div className="FavoriteItem">
					<WeatherData value={this.props.value} isMain={false}/>
				</div>
			);
		}
	}
}