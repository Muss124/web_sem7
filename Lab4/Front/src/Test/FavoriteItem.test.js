import React from 'react';
import { FavoriteItem } from '../Components/FavoriteItem/js/FavoriteItem'
import { shallow } from '../enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('Testing render FavoriteItem', function () {
    it("Loading", () => {
        const storage = {
            "city": "TEST LOADING CITY",
            "data": {}
        };
        const tree = shallow(<FavoriteItem value={storage} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("Correct data", () => {
        const storage = {
            "city": "Kiev",
            "data": {
                CityName: "Kiev",
                Temperature: 2.98,
                Weather: "light intensity drizzle",
                Humidity: 93,
                Pressure: 1014,
                Wind: 4,
                Longitude: 30.52,
                Latitude: 50.43,
                Icon: "09n",
                Cod: 200
            }
        };
        const tree = shallow(<FavoriteItem value={storage} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
});    