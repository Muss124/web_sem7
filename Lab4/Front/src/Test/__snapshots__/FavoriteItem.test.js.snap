// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render FavoriteItem Correct data 1`] = `
<div
  class="FavoriteItem"
>
  <WeatherData
    value={
      Object {
        "base": "stations",
        "clouds": Object {
          "all": 30,
        },
        "cod": 200,
        "coord": Object {
          "lat": 50.43,
          "lon": 30.52,
        },
        "dt": 1576604944,
        "id": 703448,
        "main": Object {
          "feels_like": 2.93,
          "humidity": 87,
          "pressure": 1017,
          "temp": 6.29,
          "temp_max": 7.78,
          "temp_min": 4,
        },
        "name": "Kiev",
        "sys": Object {
          "country": "UA",
          "id": 8903,
          "sunrise": 1576561989,
          "sunset": 1576590872,
          "type": 1,
        },
        "timezone": 7200,
        "visibility": 10000,
        "weather": Array [
          Object {
            "description": "scattered clouds",
            "icon": "03n",
            "id": 802,
            "main": "Clouds",
          },
        ],
        "wind": Object {
          "deg": 230,
          "speed": 3,
        },
      }
    }
  />
</div>
`;

exports[`Testing render FavoriteItem Loading 1`] = `
<div>
  TEST LOADING CITY
  <div
    class="spinner-border text-danger"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
